.github/workflows/test-docs.yml

```diff
@@ -72,3 +77,29 @@ jobs:
         run: |
           cd docs
           make linkcheck
+
+      # make rediraffecheckdiff compares files with the origin/main and other
+      # git references, and forks can be outdated, we only run these steps when
+      # working against jupyterhub/jupyterhub to avoid incorrectly erroring
+      - name: check redirects for this PR
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          cd docs
+          export REDIRAFFE_BRANCH=origin/${{ github.base_ref }}
+          make rediraffecheckdiff
+
+      # this should check currently published 'stable' links for redirects
+      - name: check redirects since last release
+        if: github.repository == 'jupyterhub/jupyterhub'
+        run: |
+          cd docs
+          export REDIRAFFE_BRANCH=$(git describe --tags --abbrev=0)
+          make rediraffecheckdiff
+
+      # longer-term redirect check (fixed version) for older links
+      - name: check redirects since 3.0.0
+        if: github.repository == 'jupyterhub/jupyterhub'
```

Comment:
```suggestion
        if: github.event_name == 'pull_request'
        run: |
          cd docs
          export REDIRAFFE_BRANCH=origin/${{ github.base_ref }}
          make rediraffecheckdiff

      # this should check currently published 'stable' links for redirects
      - name: check redirects since last release
        if: github.event_name == 'pull_request'
        run: |
          cd docs
          export REDIRAFFE_BRANCH=$(git describe --tags --abbrev=0)
          make rediraffecheckdiff

      # longer-term redirect check (fixed version) for older links
      - name: check redirects since 3.0.0
        if: github.event_name == 'pull_request'
```

Reply:
I think your conditions were correct the first time - the repo check helps ensure the tags are defined, which is relevant for the latter two checks. The PR check is only for the base ref, which doesn't exist if it's not a PR.

Reply:
Updated the comment to explain the reason for the two different `if`s (and removed the unnecessary `${{}}`, thanks!)